packages/loaders/url/src/index.ts

```diff
@@ -519,13 +519,24 @@ export class UrlLoader implements Loader<LoadFromUrlOptions> {
       https: 'wss',
       http: 'ws',
     });
+    let executorConnectionParams = {};
     const subscriptionClient = createClient({
       url: WS_URL,
       webSocketImpl,
-      connectionParams,
+      connectionParams: () => {
+        const optionsConnectionParams =
+          (typeof connectionParams === 'function' ? connectionParams() : connectionParams) || {};
+        return Object.assign(optionsConnectionParams, executorConnectionParams);
+      },
       lazy: true,
     });
     return ({ document, variables, operationName, extensions }) => {
+      // additional connection params can be supplied through the "webSocketConnectionParams" field in extensions.
+      // TODO: connection params only from the FIRST operation in lazy mode will be used (detect connectionParams changes and reconnect, too implicit?)
+      if (extensions?.['webSocketConnectionParams'] && typeof extensions?.['webSocketConnectionParams'] === 'object') {
+        executorConnectionParams = Object.assign(executorConnectionParams, extensions['webSocketConnectionParams']);
```

Comment:
This looks like leaking because everytime you call executor, it manipulates the "global" object. 
```suggestion
        const requestConnectionParams = Object.assign({}, executorConnectionParams, extensions['webSocketConnectionParams']);
```

Reply:
But it should manipulate the global object, the `executorConnectionParams` are used by the connectionParams option in graphql-ws createClient above, see here:

https://github.com/ardatan/graphql-tools/blob/7acc717c32e9160cf9c7d35cdf42942025bba9a3/packages/loaders/url/src/index.ts#L526-L530